### PR TITLE
Update documentation for logging options

### PR DESCRIPTION
**Pull request by flimzy**
I adapted the description from that found in the default.ini.

### CODE REVIEW

File: src/api/server/common.rst [hunk 1]
```diff
@@ -333,6 +333,9 @@
 .. http:get:: /_log
     :synopsis: Returns the server log file
 
+    .. warning::
+        The ``/_log`` endpoint was removed in CouchDB 2.0.0.
+
```
Comment (flimzy) [inline src/api/server/common.rst]: Or would it be better to delete this entire section?

**Comment by nickva** [comment]
Thank, you @flimzy 

I fixed a small typo in "option" (made a small commit in GH directly).

The content looks good, I let's tweak the formatting a bit. I think the option example section an extra '::' before them so they stand out when they are rendered.

For example:

```
   .. config:option:: syslog_port

        Specifies the syslog port to connect to when sending logs. Only used by
        the ``syslog`` :option:`writer <log/writer>`.

        [log]
        syslog_port = 514
```

Compared to: 

```
        The delay, in seconds, between each check for which database and view
        indexes need to be compacted::

            [compaction_daemon]
            check_interval = 300
```

Notice the '::' bit

**Comment by flimzy** [comment]
Thanks for the suggestions. And doubly thanks for the patience. This is my first foray into RST editing, so many of these subtleties don't stand out to me just yet.

**Comment by nickva** [comment]
+1

Thank you again for your contribution! This looks much better.

Will rebase and merge

**Pull request merged**